integration_tests/src/main/python/get_json_test.py

```diff
@@ -342,3 +341,42 @@ def assert_gpu_did_fallback(sql_text):
     assert_gpu_did_fallback('get_json_object(a, b)')
     assert_gpu_did_fallback('get_json_object(\'%s\', b)' % scalar_json)
 
+@pytest.mark.parametrize('json_str_pattern', [r'\{"store": \{"fruit": \[\{"weight":\d,"type":"[a-z]{1,9}"\}\], ' \
+                   r'"bicycle":\{"price":[1-9]\d\.\d\d,"color":"[a-z]{0,4}"\}\},' \
+                   r'"email":"[a-z]{1,5}\@[a-z]{3,10}\.com","owner":"[a-z]{3,8}"\}',
+                   r'\{"a": "[a-z]{1,3}"\}'], ids=idfn)
+def test_get_json_object_legacy(json_str_pattern):
+    gen = mk_json_str_gen(json_str_pattern)
+    scalar_json = '{"store": {"fruit": [{"name": "test"}]}}'
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: unary_op_df(spark, gen, length=10).selectExpr(
+            'get_json_object(a,"$.a")',
+            'get_json_object(a, "$.owner")',
+            'get_json_object(a, "$.store.fruit[0]")',
+            'get_json_object(\'%s\', "$.store.fruit[0]")' % scalar_json,
+            ),
+        conf={'spark.sql.parser.escapedStringLiterals': 'true',
+              'spark.rapids.sql.getjsonobject.legacy.enabled': 'true'})
+
+# Verify that the legacy mode is used and xfail the test which currently passes
+@pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/10218")
```

Comment:
Why are we marking a test as xfail using an issue that has been closed? Can we please have an issue that is filed to remove the legacy support instead?

Reply:
I filed https://github.com/NVIDIA/spark-rapids/issues/10680 as follow-up and remove this xfail.